src/corehost/inc/pal.h

```diff
@@ -96,7 +98,7 @@ namespace pal
 
     bool realpath(string_t& path);
     bool file_exists(const string_t& path);
-    std::vector<pal::string_t> readdir(const string_t& path);
+    void readdir(const string_t& path, std::vector<pal::string_t>* list);
```

Comment:
Pointers are out or inout params. Refs are usually const. This is because C++ doesn't have a ref keyword equivalent. Just to make it obvious. There are other places, but if I touched a function, I changed it.